wasi/wasi.go

```diff
@@ -162,6 +179,7 @@ type api struct {
 	opened map[uint32]fileEntry
 	// timeNowUnixNano is mutable for testing
 	timeNowUnixNano func() uint64
+	randSource      *rand.Rand
```

Comment:
I imagine one would expect this to be "cryptographically secure random number generator" rather than the pseudo one (See getrandom(2) on linux https://man7.org/linux/man-pages/man2/getrandom.2.html, and wasi's get_random is the counter part for that).

How about using crypto/rand instead of math.Rand?

Reply:
we can, for sure
WASI spec says "high-quality" random values, and it's somewhat confusing what they mean under this.
Will move now to crypto/rand

Reply:
yeah I agree crypto is better as many things in WASI promise things of higher security, it fits